src/librustdoc/html/format.rs

```diff
@@ -676,6 +676,36 @@ fn primitive_link(
     Ok(())
 }
 
+crate fn primitive_link_url(
+    prim: clean::PrimitiveType,
+    cx: &Context<'_>,
+    root_path: &str,
+) -> Option<String> {
+    let m = &cx.cache();
+    match m.primitive_locations.get(&prim) {
+        Some(&def_id) if def_id.is_local() => {
+            Some(format!("{}primitive.{}.html", root_path, prim.as_sym(),))
```

Comment:
Shouldn't this include the crate name? Are there any tests that the links this generates aren't giving 404s?

Reply:
Yes it should actually... I'll add one.

Reply:
Is there any way to test the link is now correct? I'm worried this will get out of sync with other parts of rustdoc.

Reply:
I added a test with a "local" primitive to check the full URL.